pkg/ds/farm_test.go

```diff
@@ -1007,6 +1007,96 @@ func TestRelock(t *testing.T) {
 	assertLabel("node3", "first farm quit")
 }
 
+func TestDieAndUpdate(t *testing.T) {
```

Comment:
unfortunately, a lot of duplicate code with that of #813. May be worth deduplicating sometime.